core/server/master/src/main/java/alluxio/master/journal/tool/RaftJournalDumper.java

```diff
@@ -81,74 +75,69 @@ void dumpJournal() throws Throwable {
+              if (proto.hasStateMachineLogEntry()) {
+                try {
+                  Journal.JournalEntry entry = Journal.JournalEntry.parseFrom(
+                      proto.getStateMachineLogEntry().getLogData().asReadOnlyByteBuffer());
+                  writeSelected(out, entry);
+                } catch (Exception e) {
+                  throw new RuntimeException(e);
+                }
+              }
+            });
+        LOG.info("Read {} entries from log {}.", entryCount, path.getPath());
       }
     } catch (Exception e) {
       LOG.error("Failed to read logs from journal.", e);
     }
   }
 
-  private void readCopycatSnapshotFromDir() {
-    Storage journalStorage = Storage.builder().withDirectory(mInputDir).build();
-    Snapshot currentSnapshot;
-    try (final SnapshotStore copycat = journalStorage.openSnapshotStore("copycat")) {
-      if (copycat.snapshots().isEmpty()) {
-        LOG.debug("No snapshot found.");
+  private File getJournalDir() {
+    return new File(mInputDir, RaftJournalSystem.RAFT_GROUP_ID.toString());
+  }
+
+  private void readRatisSnapshotFromDir() throws IOException {
+    try (RaftStorage storage = new RaftStorage(getJournalDir(),
+        RaftServerConstants.StartupOption.REGULAR)) {
+      SimpleStateMachineStorage stateMachineStorage = new SimpleStateMachineStorage();
+      stateMachineStorage.init(storage);
+      SingleFileSnapshotInfo currentSnapshot = stateMachineStorage.getLatestSnapshot();
+      if (currentSnapshot == null) {
+        LOG.debug("No snapshot found");
         return;
       }
-      currentSnapshot = copycat.currentSnapshot();
-    }
-
-    SnapshotReader snapshotReader = currentSnapshot.reader();
-    String checkpointPath = String.format("%s-%s-%s", mCheckpointsDir, currentSnapshot.index(),
-        currentSnapshot.timestamp());
-
-    LOG.debug("Reading snapshot-Id: {}", snapshotReader.readLong());
-    try (CheckpointInputStream checkpointStream =
-        new CheckpointInputStream(new SnapshotReaderStream(snapshotReader))) {
-      readCheckpoint(checkpointStream, Paths.get(checkpointPath));
-    } catch (Exception e) {
-      LOG.error("Failed to read snapshot from journal.", e);
+      final File snapshotFile = currentSnapshot.getFile().getPath().toFile();
+      String checkpointPath = String.format("%s-%s-%s", mCheckpointsDir, currentSnapshot.getIndex(),
+          snapshotFile.lastModified());
+
+      try (DataInputStream inputStream = new DataInputStream(new FileInputStream(snapshotFile))) {
+        LOG.debug("Reading snapshot-Id: {}", inputStream.readLong());
+        try (CheckpointInputStream checkpointStream = new CheckpointInputStream(inputStream)) {
+          readCheckpoint(checkpointStream, Paths.get(checkpointPath));
+        } catch (Exception e) {
+          LOG.error("Failed to read snapshot from journal.", e);
+        }
+      } catch (Exception e) {
+        LOG.error("Failed to load snapshot {}", snapshotFile, e);
```

Comment:
If you want to use the exception as the last argument, you will have to use `String.format()` to print parameters.

Reply:
IIRC in slf4j we can add one throwable in addition to the string parameters to print full exception stack: http://www.slf4j.org/faq.html#paramException

Reply:
hrmmm, that is strange. I remember our logging messages would ignore the throwable, if we were using the string parameters...